Doppler.Sap/Controllers/BusinessPartnerController.cs

```diff
@@ -0,0 +1,71 @@
+using Doppler.Sap.Models;
+using Doppler.Sap.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+
+namespace Doppler.Sap.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    [Authorize]
+    public class BusinessPartnerController
+    {
+        private readonly ILogger<BillingController> _logger;
+        private readonly IBusinessPartnerService _businessPartnerService;
+
+        public BusinessPartnerController(ILogger<BillingController> logger, IBusinessPartnerService billingService) =>
+            (_logger, _businessPartnerService) = (logger, billingService);
+
+        [HttpPost("CreateOrUpdateBusinessPartner")]
+        public async Task<IActionResult> CreateOrUpdateBusinessPartner([FromBody] DopplerUserDTO dopplerUser)
+        {
+            _logger.LogInformation($"Received user: {dopplerUser.Email}");
+            var isValidUser = VerifyUserInformation(dopplerUser);
+            if (!string.IsNullOrEmpty(isValidUser))
```

Comment:
Maybe you could add the VerifyUserinformation inside the if condition.

Reply:
Maybe you could modify the name of the variable to indicate its content type, something like verificationResultInformation, because isValidUser seems like its content is boolean (I know it's very thin...). @fgchaio I think that she cannot move the call to the "if" because then uses it to return it and to do so she should call the method again.

Reply:
you are right. My bad!

Reply:
maybe `userVerificationError`